src/Compiler/Driver/CompilerConfig.fs

```diff
@@ -91,24 +92,37 @@ let ResolveFileUsingPaths (paths, m, fileName) =
         let searchMessage = String.concat "\n " paths
         raise (FileNameNotResolved(fileName, searchMessage, m))
 
-let GetWarningNumber (m, warningNumber: string, prefixSupported) =
-    try
-        let warningNumber =
-            if warningNumber.StartsWithOrdinal "FS" then
-                if prefixSupported then
-                    warningNumber.Substring 2
-                else
-                    raise (new ArgumentException())
-            else
-                warningNumber
+let GetWarningNumber (m, numStr: string, langVersion: LanguageVersion, isCompilerOption: bool) =
+    let argFeature = LanguageFeature.ParsedHashDirectiveArgumentNonQuotes
+    let errorInvalid = Error(FSComp.SR.buildInvalidWarningNumber numStr, m)
```

Comment:
I think this should be a function producing the error on demand ("errorInvalid()") ,and not a value.

Reply:
Done